csrc/ir/internal_nodes.h

```diff
@@ -2860,6 +2860,110 @@ class ArgsortOp : public Expr {
+//! matrices.
+//!
+//! Parameters:
+//! - out: output tensor
+//! - mat1: first input tensor
+//! - mat2: second input tensor
+//! - offsets: 1D offsets tensor
+//!
+//! The offsets tensor is a 1D tensor of shape (num_groups + 1) that specifies
+//! the starting index of each group in the mat1 and mat2 tensors.
+//!
+//! Given the number of groups as G, the operation conceptually runs G matmuls.
+//! There are three configurations of grouping, reflected by ranks of input
+//! matrices:
+//!
+//! Notation 1: prefix_sum_padded_offset = numpy.cumsum([0] + offsets)
+//! Notation 2: f(i) = prefix_sum_offsets(i) : prefix_sum_offsets(i+1)
+//!             note: f(i) is a slice with length offsets[i]
+//!
+//!     Case 1: grouped k-dimension:
+//!       input rank: mat1[ m, k ] @ mat2[ k, n ] , offsets[ g ]
+//!       requires: sum(offsets) == k
+//!       math:
+//!       for i in (0...g):
+//!         out[ i, 0:m, 0:n ] = mat1[ 0:m, f(i) ] @ mat2[ f(i), 0:n ]
+//!
+//!     Case 2: grouped m-dimension:
+//!       input rank: mat1[ m, k ] @ mat2[ g, k, n ] , offsets[ g ]
+//!       requires: sum(offsets) == m
+//!       math:
+//!       for i in (0...g):
+//!         out[ f(i), 0:n ] = mat1[ f(i), 0:k ] @ mat2[ 0:k, 0:n ]
+//!
+//!     Case 3: grouped n-dimension:
+//!       input rank: mat1[ g, m, k ] @ mat2[ k, n ] , offsets[ g ]
+//!       requires: sum(offsets) == n
+//!       math:
+//!       for i in (0...g):
+//!         out[ 0:m, f(i) ] = mat1[ 0:m, 0:k ] @ mat2[ 0:k, f(i) ]
+//!
+class GroupedMmaOp : public Expr {
+ public:
+  using Expr::Expr;
+
+  GroupedMmaOp(
+      IrBuilderPasskey,
+      Val* out,
+      Val* mat1,
+      Val* mat2,
+      Val* offsets,
+      Val* scale1 = nullptr,
+      Val* scale2 = nullptr);
+
+  NVFUSER_DECLARE_CLONE_AND_CREATE
+
+  const char* getOpString() const override {
+    return "GroupedMmaOp";
+  }
+
+  std::string toString(int indent_size = 0) const override;
+  std::string toInlineString(int indent_size = 0) const override;
+  std::vector<PolymorphicValue> evaluate(
+      const ExpressionEvaluator& ee,
+      const std::vector<PolymorphicValue>& inputs) const override;
+
+  TensorView* out() const {
+    return output(0)->as<TensorView>();
+  }
+  TensorView* mat1() const {
+    return input(0)->as<TensorView>();
+  }
+  TensorView* mat2() const {
+    return input(1)->as<TensorView>();
+  }
+  TensorView* offsets() const {
+    return input(2)->as<TensorView>();
+  }
+  TensorView* scale1() const {
+    if (inputs().size() == 5) {
+      return input(3)->as<TensorView>();
+    }
+    return nullptr;
+  }
+  TensorView* scale2() const {
+    if (inputs().size() == 5) {
+      return input(4)->as<TensorView>();
+    }
+    return nullptr;
+  }
+  bool hasScale() const {
+    return attribute<bool>(0);
+  }
+
+  IterDomain* getKIDOfMat1() const;
+  IterDomain* getKIDOfMat2() const;
+
+  std::optional<IterDomain*> getGIDOfMat1() const;
```

Comment:
```suggestion
  IterDomain* getGIDOfMat1() const;
```
Would it work if we replace nullopt with nullptr?